Semaphore-OE-Client/src/test/java/com/smartlogic/ontologyeditor/examples/ModelManipulation.java

```diff
@@ -16,66 +16,66 @@
-			oeClients.add(getOEClient("semaphore5.properties"));
-			oeClients.add(getOEClient("cloud.properties"));
-		}
-		return oeClients;
-	}
-	
-	
-	protected static OEClientReadWrite getOEClient(String fileName) throws IOException, CloudException {
-		Properties properties;
-		try (FileInputStream propertiesInputStream = new FileInputStream(fileName)) {
-			properties = new Properties();
-			properties.load(propertiesInputStream);
-		} catch (IOException e) {
-			System.err.println("Error attempting to read properties from file \"config.properties\"");
-			throw e;
-		}
+  private static Collection<OEClientReadWrite> oeClients = null;
 
-		OEClientReadWrite oeClient = new OEClientReadWrite();
-		oeClient.setProxyAddress(properties.getProperty("proxy.address"));
-		oeClient.setBaseURL(properties.getProperty("base.url"));
-		oeClient.setModelUri(properties.getProperty("model.uri"));
-		oeClient.setToken(properties.getProperty("token"));		
-		oeClient.setHeaderToken(properties.getProperty("header.token"));		
-		
-		String tokenUrl = properties.getProperty("token.url");
-		String tokenKey = properties.getProperty("token.key");
-		if ((tokenUrl != null) && (tokenKey != null)) {
-			TokenFetcher tokenFetcher = new TokenFetcher(tokenUrl, tokenKey);
-			oeClient.setCloudToken(tokenFetcher.getAccessToken());
-		}
-		return oeClient;
-	}
-	
-	public static void runTests(ModelManipulation modelManipulation) throws IOException, CloudException, OEClientException {
-		Collection<OEClientReadWrite> oeClients = getOEClients();
-		for (OEClientReadWrite oeClient: oeClients) {
-			modelManipulation.alterModel(oeClient);
-		}
-	}
-	
-	public String getModelName(OEClientReadOnly oeClient) {
-		return oeClient.getModelUri().substring(oeClient.getModelUri().indexOf(":") + 1);
-	}
-	
-	protected abstract void alterModel(OEClientReadWrite oeClient) throws OEClientException;
+  public static Collection<OEClientReadWrite> getOEClients() throws IOException, CloudException {
+    if (oeClients == null) {
+      oeClients = new ArrayList<>();
+      oeClients.add(getOEClient("semaphore4.properties"));
+      oeClients.add(getOEClient("semaphore5.properties"));
+      // oeClients.add(getOEClient("cloud.properties"));
+    }
+    return oeClients;
+  }
 
-	protected String urlEncode(String value) {
-		try {
-			return URLEncoder.encode(value, "UTF-8");
-		} catch (UnsupportedEncodingException e) {
-			System.err.println("UnsupportedEncodingException - shouldn't get here");
-			return "INVALID LABEL";
-		}
-	}
+  protected static OEClientReadWrite getOEClient(String fileName)
+      throws IOException, CloudException {
+    Properties properties;
+    try (FileInputStream propertiesInputStream = new FileInputStream(fileName)) {
+      properties = new Properties();
+      properties.load(propertiesInputStream);
+    } catch (IOException e) {
+      System.err.println("Error attempting to read properties from file \"config.properties\"");

```

Comment:
Should this write the name of the file passed to getOEClient? i.e. Error attempting to read properties from file " + fileName)